keps/sig-node/4800-cpumanager-split-uncorecache/README.md

```diff
@@ -0,0 +1,689 @@
+
+![Figure to the results of uncore cache alignment](https://github.com/ajcaldelas/enhancements/blob/split_l3_cache/keps/sig-node/4800-cpumanager-split-uncorecache/images/results.PNG) ![Figure of results](https://github.com/ajcaldelas/enhancements/blob/c7e423439608001cc837e2f5629dc87935353611/keps/sig-node/4800-cpumanager-split-l3cache/images/chart.png)
+
+### Goals
+
+- Introduce a new CPU Manager policy option that assigns CPU within the same grouping of uncore cache to container scope.  
+- Minimize the number of CPUs assignments from different uncore cache grouping to pods and container scope.
+
+### Non-Goals
+
+- This proposal does not aim to modify CPU assignments for CPU Manager policy set to none.
+- This does not alter the behavior of existing static policy options such as full-pcpus-only.
+
+## Proposal
+
+- Add a new static policy option to fine-tune of CPU assignments to have a preference to align by uncore cache grouping.
+
+- Topology struct already contains CPUDetails which is a map of CPUInfo.  CPUInfo knows about NUMA, socket, and core IDs associated with a CPU.  We just need to add a new member called uncorecacheId into the CPUInfo struct that tracks whether the CPU is part of a split uncore caches. Functionality to support this was merged in cadvisor with these pull-requests [cadvisor/pr-2849](https://github.com/google/cadvisor/pull/2849) and [cadvisor/pr-2847](https://github.com/google/cadvisor/pull/2847/)
+
+- Handle enablement of the policy option in pkg/kubelet/cm/cpumanager/policy_options.go and check the validity of the user flag and capability of the system.  If topology does not support split uncore caches, grouping by uncore cache static policy will be ignored.  
+
+- Modify the "Allocate" static policy to check for the option, align-cpus-by-uncorecache. For platforms where SMT is enabled, align-cpus-by-uncorecache will continue to follow default behaviour and try to allocate full cores when possible. Align-cpus-by-uncorecache can be enabled along with full-pcpus-only and enforce full core assignment with uncorecache alignment.
+
+- Align-cpus-by-uncorecache will be compatible with the default scheduling algorithm with future plans to be compatible with the options distribute-cpus-across-numa and distribute-cpus-across-cores.
+
+### User Stories (Optional)
+
+#### Story 1
+
+- As a HPC user, I want to extract the best performance possible and reduce latency, so that I can extract the most value when deploying applications in Kubernetes. I can use the static CPU Manager Policy option, but on split uncore cache processors, my application can experience latency when assigned CPUs across multiple uncore caches. In order to maximize performance, I want to minimize the distribution across uncore caches to minimize the latency. I want the scheduler to automatically handle this for me for larger clusters for simple deployment and efficiency.
+
```

Comment:
in this context the "scheduler" is the kube-scheduler or the cpu allocation logic inside the kubelet?

Reply:
The cpu allocation logic inside the kubelet

Reply:
updated